Userland/Libraries/LibDSP/Clip.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021, kleines Filmröllchen <malu.bertsch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Clip.h"
+
+namespace LibDSP {
+
+Sample AudioClip::sample_at(u32 time)
+{
+    VERIFY(time < m_length);
+    return m_samples[time];
+}
+
+void NoteClip::set_note(RollNote note)
+{
+    VERIFY(note.pitch >= 0 && note.pitch < note_count);
+    VERIFY(note.off_sample < m_length);
+    VERIFY(note.length() >= 2);
+
```

Comment:
I would suggest `auto& notes = m_notes[note.pitch]` to increase code legibility.